athom-smart-plug-v2.yaml

```diff
@@ -84,27 +84,31 @@ sensor:
     update_interval: 60s
 
   - platform: cse7766
-    update_interval: 10s
     current:
       name: "Current"
       filters:
-          - lambda: if (x < 0.060) return 0.0; else return x;   #For the chip will report less than 3w power when no load is connected
+        - throttle_average: 10s
+        - lambda: if (x < 0.060) return 0.0; else return x;   #For the chip will report less than 3w power when no load is connected
 
 
     voltage:
       name: "Voltage"
+      filters:
+        - throttle_average: 10s
     power:
       name: "Power"
       id: power_sensor
       filters:
-          - lambda: if (x < 3.0) return 0.0; else return x;    #For the chip will report less than 3w power when no load is connected
+        - throttle_average: 10s
+        - lambda: if (x < 3.0) return 0.0; else return x;    #For the chip will report less than 3w power when no load is connected
 
 
     energy:
       name: "Energy"
       id: energy
       unit_of_measurement: kWh
       filters:
+        - throttle_average: 10s
```

Comment:
This needs to be `-  delta: 0.001`. Using `throttle_average` will cause Total Energy to break. I suspect throttle average is doing something to the value.

Reply:
Good point - I didn't notice that it broke total energy on my device. Removing throttle and adding delta: 0.001 makes energy update a lot more frequently for me, but isn't fixing Total Energy. Total Daily Energy works, though.

Reply:
> Good point - I didn't notice that it broke total energy on my device. Removing throttle and adding delta: 0.001 makes energy update a lot more frequently for me, but isn't fixing Total Energy. Total Daily Energy works, though.

Unfortunately. It wont fix it. The data becomes broken and since it is stored you have to clear it. You can factory default or you can change the globals to `restore_value` no. Flash. Then back to yes. Then flash again. That will clear the broken data. 

How often is yours updating? It would take a heavy load to make that value increment really quickly?

Reply:
For example running a 7 watt constant load updates that value in 30 seconds on my devices. 

Maybe prudent to set that delta to be .002 or .003 to slow that down.

Reply:
> It's doing about ten updates every four seconds, which is more than it needs to be.

Is that the correct interval for the amount of power you are using?

The issue I had with heartbeat is I think it will miss data. Not sure how much that matters but I think it could really add up over time because the usage in that time is lost? I could be wrong but in the past that info was not lost it just not reported to HASS until the update interval was triggered? At least with the way this value is used/calculated I believe the new filters applied will effect the data.   

I am no expert in how ESPHome works and I maybe incorrect on how this works.

Reply:
There was an issue with the `throttle_average` filter that it would give a `NaN` if there was no data.  I fixed this in the last point release.

Reply:
Why don't you just use the `throttle:` filter?  There's no data to "miss" since it's a cumulative value, so the `heartbeat` filter could work as well.  It will send the latest value on the given interval, so it's effectively the same as using an `update_interval`.